tools/building.py

```diff
@@ -1443,6 +1443,13 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
   if settings.GENERATE_SOURCE_MAP and outfile:
     cmd += [f'--input-source-map={infile}.map']
     cmd += [f'--output-source-map={outfile}.map']
+  # we should skip all validation when 'no-validation' pass is specified
+  if settings.BINARYEN_EXTRA_PASSES:
+    # BINARYEN_EXTRA_PASSES is comma-separated
+    extras = settings.BINARYEN_EXTRA_PASSES.split(',')
+    # we support both '-'-prefixed and unprefixed pass names
+    if '--no-validation' in extras or 'no-validation' in extras:
+      cmd += ['--no-validation']
```

Comment:
Is this a the right place to put `--no-validation`.. its BINARYEN_EXTRA_PASSES is, up until now, something the only effects the running of `wasm-opt`.

Perhaps a new `-s NO_VALIDATE` setting would make more sense?

Reply:
Currently, BINARYEN_EXTRA_PASSES is a single possible place to use --no-validation.

Default settings require my action:

> struct PassOptions {
  ...
  // Whether to run the validator to check for errors.
  bool validate = true;

Reply:
I noticed that we already have `BINARYEN_FEATURES` which is almost what we want here, except that it is an internal setting atm. If we moved that from `src/settings_internal.js` to `src/settings.js` then this would work.

The name is because it is used to pass around features primarily, but nothing prevents other things from being there like `--no-validation`. Perhaps we should rename it though to `BINARYEN_FLAGS`?

Reply:
I've be OK with adding `BINARYEN_FLAGS` assuming we can show this is really needed.  I think in order to make it work while we would need to show a significant amount of the link time could be saved by adding `--no-validation`.

Reply:
(My preferred option would be will be to modify `BINARYEN_PASS_DEBUG` to have a mode that didn't slow things down quite as much since that is that what is specifically needed here).

Reply:
I hope to finish tests for next weekends. Then things can come clear.